test/test_oauth2.py

```diff
@@ -206,19 +207,29 @@ def test_it_requires_code_when_exchanging(test_client: FlaskClient) -> None:
 
 
 @patch('profiles.repositories.generate_random_string')
-@responses.activate
 def test_it_exchanges(generate_random_string: MagicMock, test_client: FlaskClient) -> None:
     generate_random_string.return_value = '1a2b3c4e'
 
-    responses.add(responses.POST, 'http://www.example.com/server/token', status=200,
-                  json={'access_token': '1/fFAGRNJru1FTz70BzhT3Zg', 'expires_in': 3920,
-                        'foo': 'bar', 'token_type': 'Bearer', 'orcid': '0000-0002-1825-0097',
-                        'name': 'Josiah Carberry'})
-
-    response = test_client.post('/oauth2/token',
-                                data={'client_id': 'client_id', 'client_secret': 'client_secret',
-                                      'redirect_uri': 'http://www.example.com/client/redirect',
-                                      'grant_type': 'authorization_code', 'code': '1234'})
+    with requests_mock.Mocker() as mocker:
```

Comment:
Can't use the decorator due to https://github.com/pytest-dev/pytest/issues/2749.

Reply:
at least it shows the phase in which the mock is called